pbxproj/pbxsections/PBXShellScriptBuildPhase.py

```diff
@@ -15,7 +16,10 @@ def create(cls, script, shell_path=u"/bin/sh", files=None, input_paths=None, out
             u'shellPath': shell_path,
             u'shellScript': script,
             u'showEnvVarsInLog': show_in_log
-        })
+        }
+        if name is not None:
+            kwargs[u'name'] = name
```

Comment:
it can be simply achieved by adding the `name=None` parameter and the name unconditionally to the `parse` function. the parse function will ignore any None value from the dictionaries.